### PR TITLE
docs: move template compiler options to a separate file

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The template compiler options are in the same doc that describes AOT compiler.
Issue Number: N/A

## What is the new behavior?
 Moved template compiler options to a separate file

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/aot-compiler-options.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Angular template compiler options
+
+When you use AOT compilation, you can specify template compiler options as members of the `"angularCompilerOptions"` object in the `tsconfig.json` file to control your app compilation.
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: I tripped a bit parsing this. 

When you use AOT compilation, you can specify template compiler options as members of the `"angularCompilerOptions"` object in the `tsconfig.json` file.

or 

When you use AOT compilation, you can control your app compilation by specifying template compiler options as members of the `"angularCompilerOptions"` object in the `tsconfig.json` file.

File: aio/content/guide/aot-compiler-options.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        "preserveWhitespaces": true,
+                  ...
+      }
+  }
+  ```
+This page describes the avaialble Angular template compiler options.
+
+### *allowEmptyCodegenFiles*
+
+Tells the compiler to generate all the possible generated files even if they are empty. This option is
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: ...files, even if...

File: aio/content/guide/aot-compiler-options.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+`false` by default. This is an option used by the Bazel build rules and is needed to simplify
+how Bazel rules track file dependencies. It is not recommended to use this option outside of the Bazel
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Editing: This option is used by the Bazel build rules to simplify how Bazel rules track file dependencies. Do not use this option outside of the Bazel rules.

File: aio/content/guide/aot-compiler-options.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+rules.
+
+### *annotationsAs*
+
+Use this option to modify how the Angular specific annotations are emitted to improve tree-shaking. Non-Angular
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: Modifies how Angular-specific annotations are emitted to improve tree-shaking.

File: aio/content/guide/aot-compiler-options.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+</style>
+
+<table>
+  <tr>
+    <th>Value</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><code>decorators</code></td>
+    <td>Leave the decorators in place. This makes compilation faster. TypeScript will emit calls to the __decorate helper.  Use <code>--emitDecoratorMetadata</code> for runtime reflection.  However, the resulting code will not properly tree-shake.</td>
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed to remove future tense: Causes TypeScript to emit calls to the __decorate helper.  Use <code>--emitDecoratorMetadata</code> for runtime reflection.  Note: the resulting code does not properly tree-shake.

Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: Leaves the decorators in place. 
(Judy started using this style in API docs and CLI ref, so let's continue it. Thx.)

File: aio/content/guide/aot-compiler-options.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  </tr>
+  <tr>
+    <td><code>static fields</code></td>
+    <td>Replace decorators with a static field in the class. Allows advanced tree-shakers like
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: Replaces...

File: aio/content/guide/aot-compiler-options.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    <a href="https://github.com/google/closure-compiler">Closure compiler</a> to remove unused classes.</td>
+  </tr>
+  </table>
+
+
+### *annotateForClosureCompiler*
+
+This option tells the compiler to use [Tsickle](https://github.com/angular/tsickle) to annotate the emitted
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Another style match to API and CLI refs: "Tells the...

File: aio/content/guide/aot-compiler-options.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+The Angular template compiler transforms code that is used, or could be used, in an annotation
+to allow it to be imported from template factory modules. See
+[metadata rewriting](guide/aot-compiler#metadata-rewriting) for more information.
+
+Setting this option to `false` disables this rewriting, requiring the rewriting to be
+done manually.
+
+### *disableTypeScriptVersionCheck*
+
+When `true`, this option tells the compiler not to check the TypeScript version.
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: When `true`, instructs the compiler
Everywhere: I like "instructs the compiler to" or "causes the compiler to" better than "tells..."

File: aio/content/guide/aot-compiler-options.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+      "angularCompilerOptions": {
+        "fullTemplateTypeCheck": true,
+        "preserveWhitespaces": true,
+                  ...
+      }
+  }
+  ```
+This page describes the avaialble Angular template compiler options.
+
+### *allowEmptyCodegenFiles*
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: There was a previous style that used italic in titles for things that would normally be code font. I think we have standardized on using the regular font, so please remove the italics. Thx. (I think sometimes we do use code font in headings, but only where we need to differentiate it from other words in the heading.) (Style ref: https://developers.google.com/style/headings)

File: aio/content/guide/aot-compiler-options.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+### *disableTypeScriptVersionCheck*
+
+When `true`, this option tells the compiler not to check the TypeScript version.
+The compiler will skip checking and will not error out when an unsupported version of TypeScript is used.
+Setting this option to `true` is not recommended because unsupported versions of TypeScript might have undefined behaviour.
+
+This option is `false` by default.
+
+### *enableResourceInlining*
+This option instructs the compiler to replace the `templateUrl` and `styleUrls` property in all `@Component` decorators with inlined contents in `template` and `styles` properties.
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: "...`templateUrl` and `styleUrls` properties..."

File: aio/content/guide/aot-compiler-options.md [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+if the metadata is used to determine the contents of an annotation. The metadata
+collector cannot predict the symbols that are designed for use in an annotation, so it will preemptively
+include error nodes in the metadata for the exported symbols. The template compiler can then use the error
+nodes to report an error if these symbols are used. If the client of a library intends to use a symbol in an annotation, the template compiler will not normally report
+this until the client uses the symbol. This option allows detecting these errors during the build phase of
+the library and is used, for example, in producing Angular libraries themselves.
+
+### *strictInjectionParameters*
+
+When set to `true`, this options tells the compiler to report an error for a parameter supplied
```
Comment (jenniferfell) [inline aio/content/guide/aot-compiler-options.md]: Ed: When set to `true`, this option...

**Review by jenniferfell — CHANGES_REQUESTED** [review]
One navigation change. 

Misc editorial recommendations. In some cases I only marked the first occurrence, not all of them. 
Although we try to avoid future tense, if you can't make something clear in present tense, future isn't a deal breaker. ("When this option is true the compiler does x" is better than "...the compiler will do x." 

I have a few technical questions in a separate comment. One could affect the filename and titles.

**Comment by jenniferfell** [comment]
Internal team note: https://angular-team.atlassian.net/browse/DOCS-479

File: aio/content/guide/aot-compiler-options.md (hunk 5 above)
Comment (IgorMinar) [inline aio/content/guide/aot-compiler-options.md]: please put `__decorate` into backticks:
```
`__decorate`
```

**Review by IgorMinar — COMMENTED** [review]
Can we call the file angular-compiler-options.md since all of these config options are in the `angularCompilerOptions` config block of tsconfig.json.

We should go with `angularCompilerOptions.md` instead of `angular-compiler-options.md` unless we have other precedence for case sensitive file names in the doc corpus (I'm not aware of any). Case sensitive filenames are often source of cross-platform compatibility issues (windows vs mac vs linux) so it's better to avoid them.

**Comment by jbogarthyde** [comment]
Transfer content to https://github.com/angular/angular/pull/31829

**Pull request closed without merging**